#!/usr/bin/env python
PACKAGE = "bobi_vision"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("num_agents",    int_t,    0, "Number of individuals to track", 2,  0, 100)
gen.add("num_robots",    int_t,    0, "Number of robots to track", 1,  0, 5)
method_enum = gen.enum([ gen.const("None",      int_t, 0, "No filtering"),
                       gen.const("NearestCentroids",     int_t, 1, "Nearest past moment")],
                     "Trajectory Identification Method")
gen.add("method", int_t, 0, "Trajectory Identification Method", 0, 1, 2, edit_method=method_enum)


exit(gen.generate(PACKAGE, "trajectory_identification_node", "TrajectoryIdentification"))